clients/client-python/gravitino_client/core/dto.py

```diff
@@ -0,0 +1,10 @@
+"""
+Copyright 2024 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2.
+"""
+from pydantic import BaseModel
+
+class VersionDTO(BaseModel):
+    version: str
+    compile_date: str
+    git_commit: str
```

Comment:
Since the HTTP payload is the really simple data structure, the embedded [json](https://docs.python.org/3/library/json.html#module-json) or [Data Classes](https://docs.python.org/3/library/dataclasses.html) package should handle it more straightforwardly. e.g.

```suggestion
from dataclasses import dataclass

@dataclass
class VersionDTO:
    version: str
    compile_date: str
    git_commit: str
```

After that, the `pydantic` might be removed from depending packages.

Reply:
It indeed simplify the dependency, thanks for sharing!

However, it might require us to remind the package user to use python version up to at least 3.8?

Reply:
Yes, so we should define a minimal version of cPython.

Reply:
Should this detail be included in the README? 
Or is there a better way to communicate it?

Reply:
Maybe leave the requirement in the README is clear enough

Reply:
@Lanznx we should define at here https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#python-requirement

Reply:
OK, so the definition of python version might relate to https://github.com/datastrato/gravitino/issues/2230

Reply:
Yep, let's do it in a separated PR.

Reply:
> OK, so the definition of python version might relate to https://github.com/datastrato/gravitino/issues/2230

cc @SophieTech88